functorch/csrc/BatchRulesRandomness.cpp

```diff
@@ -399,6 +425,11 @@ TORCH_LIBRARY_IMPL(aten, FuncTorchVmapMode, m) {
   UNARY_POINTWISE_RANDOM(poisson);
   UNARY_POINTWISE_RANDOM(bernoulli);
 
+  DROPOUT_UNSQUEEZE(dropout);
+  DROPOUT_UNSQUEEZE(feature_dropout);
+  DROPOUT_UNSQUEEZE(alpha_dropout);
+  DROPOUT_UNSQUEEZE(feature_alpha_dropout);
```

Comment:
Okay, I thought about it some more. Let's talk about the case where input isn't batched but the user wanted different randomness.

So assume input has shape (3,) and we are vmapping over a dimension of size B.

For `at::dropout(input)`:
- at::native_dropout(input) route: This is fine. We expand the input to (B, 3) and then pass it through at::native_dropout.
- for `_dropout` route: we've got a problem, https://github.com/pytorch/pytorch/blob/97b20b9b50b5b1460c7284438184901c6eecaa3c/aten/src/ATen/native/Dropout.cpp#L58-L59. at::empty_like(input) produces a Tensor of size (3,), but then we call `bernoulli_` on it :/. This raises an error because we cannot do in-place on this randomness pattern.

Am I analyzing this correctly? If so, this implies we'll need to change pytorch/pytorch's implementation of at::dropout, perhaps by introducing an out-of-place at::bernoulli operation.